src/python/pants/engine/internals/engine_test.py

```diff
@@ -603,14 +603,21 @@ async def a_rule() -> TrueResult:
         )
 
         with handler.session():
+            scheduler.record_test_observation(128)
             scheduler.product_request(TrueResult, subjects=[0])
+            histograms_info = scheduler.get_observation_histograms()
 
         finished = list(itertools.chain.from_iterable(tracker.finished_workunit_chunks))
         workunits_with_counters = [item for item in finished if "counters" in item]
         assert workunits_with_counters[0]["counters"]["local_execution_requests"] == 1
         assert workunits_with_counters[1]["counters"]["local_cache_requests"] == 1
         assert workunits_with_counters[1]["counters"]["local_cache_requests_uncached"] == 1
 
+        assert histograms_info["version"] == 0
+        assert "histograms" in histograms_info
+        assert "test_observation" in histograms_info["histograms"]
+        assert len(histograms_info["histograms"]["test_observation"]) > 0
```

Comment:
those assertions should be tighter. assert on the actual data not only on length and keys.
ideally we should have `assert histograms_info == {..........}`

Reply:
The only histogram guaranteed to be in this test is `test_observation` which is triggered by this test. Asserting that there are no other histograms requires this test to know about what other metrics get added over time.

I did add the bytes for the encoded histogram as per your suggestion to assert on data. But that is only for `test_observation`.)